src/ErrorHandler.php

```diff
@@ -19,29 +12,322 @@
      * Raven client.
      *
      * @param callable $callback           callback that will be called when exception is caught
      * @param int      $reservedMemorySize The amount of memory to reserve for the fatal error handler
      *
-     * @throws \Exception|\Throwable|\ErrorException
-     *
-     * @return ErrorHandler
+     * @return self
      */
-    public static function register(callable $callback, $reservedMemorySize = 10240)
+    public static function register(callable $callback, int $reservedMemorySize = 10240): self
     {
         return new self($callback, $reservedMemorySize);
     }
 
     /**
-     * {@inheritdoc}
+     * Sets the PHP error levels that will be captured by the Raven client when
+     * a PHP error occurs.
+     *
+     * @param int  $levels  A bit field of E_* constants for captured errors
+     * @param bool $replace Whether to replace or amend the previous value
+     *
+     * @return int The previous value
      */
-    protected function doHandleException($exception)
+    public function captureAt(int $levels, bool $replace = false): int
+    {
+        $prev = $this->capturedErrors;
+
+        $this->capturedErrors = $levels;
+
+        if (!$replace) {
+            $this->capturedErrors |= $prev;
+        }
+
+        $this->reRegister($prev);
+
+        return $prev;
+    }
+
+    /**
+     * Handles errors by capturing them through the Raven client according to
+     * the configured bit field.
+     *
+     * @param int    $level   The level of the error raised, represented by one
+     *                        of the E_* constants
+     * @param string $message The error message
+     * @param string $file    The filename the error was raised in
+     * @param int    $line    The line number the error was raised at
+     *
+     * @return bool If the function returns `false` then the next error handler
+     *              in the chain will be called
+     *
+     * @throws \Throwable
+     *
+     * @internal
+     */
+    public function handleError(int $level, string $message, string $file, int $line): bool
+    {
+        $shouldCaptureError = (bool) ($this->capturedErrors & $level);
+
+        if (!$shouldCaptureError) {
+            return false;
+        }
+
+        $errorAsException = new \ErrorException(self::ERROR_LEVELS_DESCRIPTION[$level] . ': ' . $message, 0, $level, $file, $line);
+        $backtrace = $this->cleanBacktraceFromErrorHandlerFrames($errorAsException->getTrace(), $file, $line);
+
+        $this->exceptionReflection->setValue($errorAsException, $backtrace);
+
+        try {
+            $this->handleException($errorAsException);
+        } catch (\Throwable $exception) {
+            // Do nothing as this error handler should be as transparent as possible
+        }
+
+        if (null !== $this->previousErrorHandler) {
+            return \call_user_func($this->previousErrorHandler, $level, $message, $file, $line);
```

Comment:
As suggested in https://github.com/getsentry/sentry-php/pull/699#issuecomment-439650920
```suggestion
            return false !== \call_user_func($this->previousErrorHandler, $level, $message, $file, $line);
```

Reply:
We should also add a unit test for this.

Reply:
Done. For the moment I've added a unit test in PHPUnit but I was thinking about transforming all these tests into PHPT tests so that we can make the `handle*` methods of the error handler `private` and stop having to rely to be `public` just to test them.